Context, Params and Variables guides (#2442)
astronomer/docs: learn/airflow-params.md

```python
"my_string_param": Param(
    "Airflow is awesome!",
    type="string",
    title="Favorite orchestrator:",
    description="Enter your favorite data orchestration tool.",
    section="Important params",
    min_length=1,
    max_length=200,
)
```

![String param example](/img/guides/airflow-params_string_param_example.png)

By defining a [valid date, datetime or time format](https://datatracker.ietf.org/doc/html/rfc3339#section-5.6) you can create a calendar picker in the Trigger DAG UI.

```python
"my_datetime_param": Param(
    "2016-10-18T14:00:00+00:00",
    type="string",
    format="date-time",
),
```

![Datetime param example](/img/guides/airflow-params_datetime_picker.png)

Providing a list of values to the `enum` attribute will create a dropdown menu in the Trigger DAG UI. Note that the default value must also be in the list of valid values provided to `enum`. Due to JSON validation rules, a value has to be selected.

```python
"my_enum_param": Param(
    "Hi :)", type="string", enum=["Hola :)", "Hei :)", "Bonjour :)", "Hi :)"]
),
```

![Enum param example](/img/guides/airflow-params_enum_example.png)

> A boolean type param will create a toggle in the Trigger DAG UI.

Author: I standardized on **Trigger DAG w/config** UI, hope that works :)

```python
 "my_bool_param": Param(True, type="boolean"),
```

![Bool param example](/img/guides/airflow-params_bool.png)

> Lastly, by providing custom HTML to the `description_html` attribute you can create more complex UI elements like a color picker. You can find the code which created the UI element below in [this example DAG in the Airflow documentation](https://airflow.apache.org/docs/apache-airflow/stable/_modules/airflow/example_dags/example_params_ui_tutorial.html).

Reviewer: ```suggestion
If you provide custom HTML to the `description_html` attribute, you can create more complex UI elements like a color picker. For sample code, see the [the example DAG in the Airflow documentation](https://airflow.apache.org/docs/apache-airflow/stable/_modules/airflow/example_dags/example_params_ui_tutorial.html).
```

- `section`: Creates a section under which the param will be displayed in the Trigger DAG UI. All params with no specified section will be displayed under the default section **DAG conf Parameters**.
- `format`: a [JSON format](https://json-schema.org/draft/2020-12/json-schema-validation.html#name-dates-times-and-duration) the input will be validated against.
- `enum`: A defined list of valid values for a param. Setting this attribute will create a dropdown menu in the Trigger DAG UI.
- `const`: Setting this attribute to a valid value will make that value a constant default and hide the param from the Trigger DAG UI. Note that you still need to provide a `default` value for the param.
- `custom_html_form`: Allows you to create custom HTML on top of the provided features.

All Param attributes are optional to set. For Params of the type string you can additionally set `min_length` and `max_length` to define the minimum and maximum length of the input. Similarly Params of the types `integer` and `number` can have a `minimum` and `maximum` value.

> ### Param default examples

Author: yeah, me neither! The color picker is the one where I thought: how was this just hidden somewhere in an example DAG in the official docs, that is so cool. 😄
